src/Model/ModelManager.php

```diff
@@ -339,17 +339,13 @@ public function getIdentifierValues($entity)
                 continue;
             }
 
-            if (method_exists($value, '__toString')) {
-                $identifiers[] = (string) $value;
-
-                continue;
-            }
-
             $fieldType = $metadata->getTypeOfField($name);
             $type = $fieldType && Type::hasType($fieldType) ? Type::getType($fieldType) : null;
-            if ($type) {
+            if ($type and 'binary' === $platform->getDoctrineTypeMapping($fieldType)) {
```

Comment:
Please, try to simplify these checks:

```php
if ($type) {
    $identifiers[] = 'binary' === $platform->getDoctrineTypeMapping($fieldType) ? (string) $type->convertToPHPValue($value, $platform) : $type->convertToDatabaseValue($value, $platform);
 
    continue;
}
```

Reply:
Done.

Reply:
Please, respect CS by leaving blankspaces around ternary operators (`?` and `:`).

Reply:
Done